cypress/e2e/read-receipts/high-level.spec.ts

```diff
@@ -2344,12 +2369,236 @@ describe("Read receipts", () => {
+            assertUnreadThread("Root1");
+            assertUnreadThread("Root2");
+            assertUnreadThread("Root3");
+        });
+        it("Looking in thread view to find old threads that were never read makes the room unread", () => {
+            // Given lots of messages in threads that are unread
+            goTo(room1);
+            receiveMessages(room2, [
+                "Root1",
+                "Root2",
+                "Root3",
+                ...manyThreadedOff("Root1", many("T", 2)),
+                ...manyThreadedOff("Root2", many("T", 2)),
+                ...manyThreadedOff("Root3", many("T", 2)),
+                ...many("Msg", 100),
+            ]);
+            goTo(room2);
+            assertUnread(room2, 6);
+            assertUnreadThread("Root1");
+            assertUnreadThread("Root2");
+            assertUnreadThread("Root3");
+
+            // When I restart
+            closeThreadsPanel();
+            goTo(room1);
+            saveAndReload();
+
+            // Then the room remembers it's unread
+            // TODO: I (andyb) think this will fall in an encrypted room
+            assertUnread(room2, 6);
+
+            // And when I open the threads view
+            goTo(room2);
+            openThreadList();
+
+            // Then the room remains unread
+            assertUnread(room2, 6);
+            assertUnreadThread("Root1");
+            assertUnreadThread("Root2");
+            assertUnreadThread("Root3");
+        });
+        it("After marking room as read, paging up to find old threads that were never read leaves the room read", () => {
+            // Given lots of messages in threads that are unread but I marked as read on a main timeline message
+            goTo(room1);
+            receiveMessages(room2, [
+                "Root1",
+                "Root2",
+                "Root3",
+                ...manyThreadedOff("Root1", many("T", 2)),
+                ...manyThreadedOff("Root2", many("T", 2)),
+                ...manyThreadedOff("Root3", many("T", 2)),
+                ...many("Msg", 100),
+            ]);
+            markAsRead(room2);
+            assertRead(room2);
+
+            // When I restart
+            saveAndReload();
+
+            // Then the room remembers it's read
+            assertRead(room2);
+
+            // And when I page up to load old thread roots
+            goTo(room2);
+            pageUp();
+            pageUp();
+            pageUp();
+
+            // Then the room remains read
+            assertStillRead(room2);
+            assertReadThread("Root1");
+            assertReadThread("Root2");
+            assertReadThread("Root3");
+        });
+        // XXX: fails because we see a dot instead of an unread number - probably the server and client disagree
+        it.skip("After marking room as read based on a thread message, opening threads view to find old threads that were never read leaves the room read", () => {
```

Comment:
I assume all the skipped tests are what we need to fix?

Reply:
Yes, everything with "XXX: fails" is a bug we need to fix.